test/UnitTests/MSTest.Analyzers.UnitTests/AssemblyInitializeShouldBeValidAnalyzerTests.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Framework;
+using Microsoft.Testing.TestInfrastructure;
+
+using VerifyCS = MSTest.Analyzers.Test.CSharpCodeFixVerifier<
+    MSTest.Analyzers.AssemblyInitializeShouldBeValidAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace MSTest.Analyzers.Test;
+
+[TestGroup]
+public sealed class AssemblyInitializeShouldBeValidAnalyzerTests(ITestExecutionContext testExecutionContext) : TestBase(testExecutionContext)
+{
+    public async Task WhenAssemblyInitializeIsPublic_NoDiagnostic()
+    {
+        var code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [AssemblyInitialize]
+                public static void AssemblyInitialize(TestContext context)
+                {
+                }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(code);
+    }
+
+    public async Task WhenAssemblyInitializeIsNotOrdinary_Diagnostic()
+    {
+        var code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [AssemblyInitialize]
+                ~{|#0:MyTestClass|}()
+                {
+                }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(
+            code,
+            VerifyCS.Diagnostic(AssemblyInitializeShouldBeValidAnalyzer.OrdinaryRule)
+                .WithLocation(0)
+                .WithArguments("Finalize"));
+    }
+
+    public async Task WhenAssemblyInitializeIsGeneric_Diagnostic()
+    {
+        var code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [AssemblyInitialize]
+                public static void {|#0:AssemblyInitialize|}<T>(TestContext context)
+                {
+                }
+            }
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(
+            code,
+            VerifyCS.Diagnostic(AssemblyInitializeShouldBeValidAnalyzer.NotGenericRule)
+                .WithLocation(0)
+                .WithArguments("AssemblyInitialize"));
+    }
+
+    [Arguments("protected")]
+    [Arguments("internal")]
+    [Arguments("internal protected")]
+    [Arguments("private")]
+    public async Task WhenAssemblyInitializeIsNotPublic_Diagnostic(string accessibility)
```

Comment:
Please add either another class in this test or a different test that checks the resultant visiblity. Something like `internal class` and `public static void` method.

This way if we ever do a change and drop resultant visibility we would have a test failing.

In case I missed it in `TestInit` and `TestCleanup` PRs, would it be ok for you to do another PR also adding the check?!

Reply:
here we go: https://github.com/microsoft/testfx/pull/2346